lucene/core/src/test/org/apache/lucene/util/TestBytesRefHash.java

```diff
@@ -267,6 +271,71 @@ public void testFind() throws Exception {
     }
   }
 
+  @Test
+  public void testConcurrentAccessToUnmodifiableBytesRefHash() throws Exception {
+    int num = atLeast(2);
+    for (int j = 0; j < num; j++) {
+      int numStrings = 797;
+      List<String> strings = new ArrayList<>(numStrings);
+      for (int i = 0; i < numStrings; i++) {
+        final String str = TestUtil.randomRealisticUnicodeString(random(), 1, 1000);
+        hash.add(new BytesRef(str));
+        assertTrue(strings.add(str));
+      }
+      int hashSize = hash.size();
+
+      UnmodifiableBytesRefHash unmodifiableHash = new UnmodifiableBytesRefHash(hash);
+
+      AtomicInteger notFound = new AtomicInteger();
+      AtomicInteger notEquals = new AtomicInteger();
+      AtomicInteger wrongSize = new AtomicInteger();
+      int numThreads = 10;
```

Comment:
Can we tone this down to use less threads (at least locally, it is ok to increase for NIGHTLY). Using many threads causes tests like this to run extremely slow on wimpier machine (e.g. i have 2 cores)

Reply:
```suggestion
      int numThreads = atLeast(3);
```
Sure. I've copy&pasted it from another test. I think atLeast(3) should be fine. It's fast on smaller number of CPUs and grows to larger number of threads when nightly tests are enabled. What you think?